codalab/apps/teams/forms.py

```diff
@@ -89,3 +95,144 @@ def clean(self):
+                        name_list.append(member.username)
+                        name_text_list = ",".join(name_list)
+                    self.initial['text_members'] = name_text_list
+
+    def clean_text_members(self):
+        if self.cleaned_data.get('text_members'):
+            team_members_text = self.cleaned_data.get('text_members')
+            team_members_list = team_members_text.replace(' ','').split(',')
+            # team_members_list = team_members_text.replace(' ','').split('\n')
+            self.user_list = ClUser.objects.filter(Q(username__in=team_members_list) | Q(email__in=team_members_list))
+
+    def clean_name(self):
+        if self.cleaned_data.get('name'):
+            if hasattr(self, 'instance'):
+                if self.instance.pk:
+                    existing_team_with_name = Team.objects.filter(name=self.cleaned_data.get('name')).exclude(pk=self.instance.pk)
+                    if existing_team_with_name:
+                        raise ValidationError(('Invalid value for name. A team already exists with that name.'),
+                                              code='invalid')
+            else:
+                existing_team_with_name = Team.objects.filter(name=self.cleaned_data.get('name'))
+                if existing_team_with_name:
+                    raise ValidationError(('Invalid value for name. A team already exists with that name.'), code='invalid')
+        return self.cleaned_data.get('name')
+
+    def save(self, commit=True):
+        self.instance.creator = self.creator
+        self.instance.competition = self.competition
+        self.instance.status = TeamStatus.objects.get(codename=TeamStatus.APPROVED)
+
+        # We need to call save before adding members
+        self.instance.save()
+
+        if hasattr(self, 'user_list'):
+            # On save, if we have existing objects, clear it. We only add names in the form.
+            if len(self.instance.members.all()) > 0:
+                # Remove users no longer in the list
+                for user in self.instance.members.all():
+                    if user not in self.user_list:
+                        # Delete members not on the new list
+                        print("Deleting membership for user: {}".format(user))
+                        TeamMembership.objects.get(user=user, team=self.instance).delete()
+            # Then we loop through the names/emails they gave us.
+            for user in self.user_list:
+                if user not in self.instance.members.all():
+                    # There is not already an existing membership for this user.
+                    new_team_membership = TeamMembership.objects.create(
+                        user=user,
+                        team=self.instance,
+                        is_invitation=False,
+                        is_request=False,
+                        start_date=timezone.now(),
+                        end_date=timezone.now() + datetime.timedelta(days=9000),
```

Comment:
weird, can we just have a `null` `end_date`?

Reply:
Ah yes, null and blank are both True on that field.